azurerm/resource_arm_front_door.go

```diff
@@ -0,0 +1,1460 @@
+			},
+		}
+
+		output = append(output, result)
+	}
+
+	return &output
+}
+
+func expandArmFrontDoorBackend(input []interface{}) *[]frontdoor.Backend {
+	if len(input) == 0 {
+		return &[]frontdoor.Backend{}
+	}
+
+	output := make([]frontdoor.Backend, 0)
+
+	for _, be := range input {
+		backend := be.(map[string]interface{})
+
+		address := backend["address"].(string)
+		hostHeader := backend["host_header"].(string)
+		enabled := backend["enabled"].(bool)
+		httpPort := int32(backend["http_port"].(int))
+		httpsPort := int32(backend["https_port"].(int))
+		priority := int32(backend["priority"].(int))
+		weight := int32(backend["weight"].(int))
+
+		result := frontdoor.Backend{
+			Address:           utils.String(address),
+			BackendHostHeader: utils.String(hostHeader),
+			EnabledState:      expandArmFrontDoorBackendEnabledState(enabled),
+			HTTPPort:          utils.Int32(httpPort),
+			HTTPSPort:         utils.Int32(httpsPort),
+			Priority:          utils.Int32(priority),
+			Weight:            utils.Int32(weight),
+		}
+
+		output = append(output, result)
+	}
+
+	return &output
+}
+
+func expandArmFrontDoorBackendEnabledState(isEnabled bool) frontdoor.BackendEnabledState {
+	if isEnabled {
+		return frontdoor.Enabled
+	}
+
+	return frontdoor.Disabled
+}
+
+func expandArmFrontDoorBackendPoolsSettings(enforceCertificateNameCheck bool) *frontdoor.BackendPoolsSettings {
+	enforceCheck := frontdoor.EnforceCertificateNameCheckEnabledStateDisabled
+
+	if enforceCertificateNameCheck {
+		enforceCheck = frontdoor.EnforceCertificateNameCheckEnabledStateEnabled
+	}
+
+	result := frontdoor.BackendPoolsSettings{
+		EnforceCertificateNameCheck: enforceCheck,
+	}
+
+	return &result
+}
+
+func expandArmFrontDoorFrontendEndpoint(input []interface{}, subscriptionId string, resourceGroup string, frontDoorName string) *[]frontdoor.FrontendEndpoint {
+	if len(input) == 0 {
+		return &[]frontdoor.FrontendEndpoint{}
+	}
+
+	output := make([]frontdoor.FrontendEndpoint, 0)
+
+	for _, frontendEndpoints := range input {
+		frontendEndpoint := frontendEndpoints.(map[string]interface{})
+
+		hostName := frontendEndpoint["host_name"].(string)
+		isSessionAffinityEnabled := frontendEndpoint["session_affinity_enabled"].(bool)
+		sessionAffinityTtlSeconds := int32(frontendEndpoint["session_affinity_ttl_seconds"].(int))
+		customHttpsConfiguration := frontendEndpoint["custom_https_configuration"].([]interface{})
+		name := frontendEndpoint["name"].(string)
+		id := utils.String(helper.GetFrontDoorSubResourceId(subscriptionId, resourceGroup, frontDoorName, "FrontendEndpoints", name))
+
+		sessionAffinityEnabled := frontdoor.SessionAffinityEnabledStateDisabled
+
+		if isSessionAffinityEnabled {
+			sessionAffinityEnabled = frontdoor.SessionAffinityEnabledStateEnabled
+		}
+
+		result := frontdoor.FrontendEndpoint{
+			ID:   id,
+			Name: utils.String(name),
+			FrontendEndpointProperties: &frontdoor.FrontendEndpointProperties{
+				// ResourceState:
```

Comment:
What is the point of these comments 🤔

Reply:
I put them there to show which attributes are in the structure that I have omitted from terraform, it made it easier to keep straight in my head when I was comparing the SDK structs to what I implemented within Terraform.

Reply:
I suggest either removing them or adding a comment to that effect then

Reply:
I have removed them.